FF125 WAS multi memory ff125 (#32919)
mdn/content: files/en-us/webassembly/reference/memory/grow/index.md

---
title: "grow: WASM text instruction"
short-title: grow
slug: WebAssembly/Reference/Memory/Grow
page-type: webassembly-instruction
browser-compat:
  - webassembly.api.Memory.grow
  - webassembly.multiMemory
---

{{WebAssemblySidebar}}

The **`grow`** [memory instruction](/en-US/docs/WebAssembly/Reference/Memory) increases the size of the memory instance by a specified number of pages.

The instruction returns the previous size of memory, in pages, if the operation was successful, and `-1` if the operation failed.

> In both cases the result is added to the stack.

Author: OK. I might make it even more explicit

> The instruction returns the previous size of memory, in pages, if the operation was successful, and `-1` if the operation failed.
> In both cases the result is added to the stack.

Author: ```suggestion
The instruction adds the previous size of memory (in pages) to the top of the stack if the operation was successful, or `-1` if the operation failed.
```

;; the top item on the stack will now either be the previous number of pages (success) or `-1` (failure)
```

## Instructions and opcodes

| Instruction   | Binary opcode |
| ------------- | ------------- |
| `memory.grow` | `0x40`        |

## Browser compatibility

> > **Note:** `grow` support in WASM modules matches the grow support in the JavaScript API.

Author: FYI this is a partial update to `grow` but this statement still needs to be "broadly confirmed". It is a reasonable assumption IMO, which is why I have included the compat information for `Memory.grow()` JS API. I think it is useful - even though they are different APIs

Author: Cool. The right thing to do would be to add a BCD entry for the wasm. But we don't have those and it's a big leap.
